server/gitlab.go

```diff
@@ -141,37 +139,81 @@ func (s *Server) checkPipelinesForSameEnvs(ctx context.Context, info *E2ETestTri
 }
 
 func hasSameEnvs(info *E2ETestTriggerInfo, glVars []*gitlab.PipelineVariable) (bool, error) {
-	if info.EnvVars == nil {
-		matching := 2
-		i := 0
+	isSame, err := isPipelineForSamePRWithoutOptions(info, glVars)
+	if err != nil {
+		return false, err
+	}
+	if isSame {
+		return true, nil
+	}
+
+	isSame, err = isPipelineForSamePRWithSameOptions(info, glVars)
+	if err != nil {
+		return false, err
+	}
+	return isSame, nil
+}
+
+// The following is a sample response of retrieving pipeline environment variables of one of our e2e testing pipelines via api.
+// The first element are custom options, the rest of the elements, every pipeline has.
+// [
+// {
+// "key": "MM_ENV",
+// "value": "MM_FEATUREFLAGS_GRAPHQL=true"
+// },
+// {
+// "key": "PR_NUMBER",
+// "value": "10857"
+// },
+// {
+// "key": "REF_MATTERMOST_SERVER",
+// "value": "master"
+// },
+// {
+// "key": "REF_MATTERMOST_WEBAPP",
+// "value": "update-cypress-report"
+// },
+// {
+// "key": "SHA_MATTERMOST_SERVER",
+// "value": ""
+// },
+// {
+// "key": "SHA_MATTERMOST_WEBAPP",
+// "value": "c4a1b4cc4a7fdce4e007e95c840e59e0d976e3f3"
+// }
+// ]
+func isPipelineForSamePRWithoutOptions(info *E2ETestTriggerInfo, glVars []*gitlab.PipelineVariable) (bool, error) {
+	defaultTriggerEnvVars := [5]string{
+		"PR_NUMBER",
+		"REF_MATTERMOST_SERVER",
+		"SHA_MATTERMOST_SERVER",
+		"REF_MATTERMOST_WEBAPP",
+		"SHA_MATTERMOST_WEBAPP",
```

Comment:
These are also defined as constants in the same file, is there any reason not to reuse those instead?
Same for in `defaultEnvs` definition

Reply:
noice, thanks!